vortex-serde/src/layouts/read/mod.rs

```diff
@@ -53,56 +56,27 @@ pub enum ReadResult {
     Batch(Array),
 }
 
+#[derive(Debug)]
+pub enum RangeResult {
+    ReadMore(Vec<Message>),
+    Rows(Option<RowSelector>),
+}
+
 pub trait LayoutReader: Debug + Send {
+    /// Produce sets of row ranges to read from underlying layouts.
+    ///
+    /// Empty RangeResult indicates layout is done producing ranges
+    fn next_range(&mut self) -> VortexResult<RangeResult>;
```

Comment:
IIRC, we intend for this function to take a RowSelection in order to implement `take` pruning?

I wonder if, in the absence of a bit mask, this function should still take a row_offset to start from? Otherwise all layouts everywhere have to maintain some internal state about the current row offset..

We could then remove the `advance` function and say that the protocol is for next_range to never go backwards, i.e. row_offset `>=` the previous row_offset.